quinn/src/connection.rs

```diff
@@ -258,6 +258,9 @@ impl Connection {
 }
 
 /// A stream of QUIC streams initiated by a remote peer.
+///
+/// Incoming streams are opened in the same order that the peer created them, but data can be
```

Comment:
The former part of this sentence doesn't seem completely robust in case of datagram reordering, right?

Reply:
This is actually exactly correct as written, which is why it's worth calling out explicitly. Stream IDs are contiguous (modulo type bits), so if we see a discontinuity it's guaranteed that it's due to loss or reordering and the peer intended to open other streams first. This allows application protocols to assign special meaning to certain streams based on ordering without having to add tags, at the cost of requiring applications to take care not to do all stream processing in-order.